src/server/api/routers/traces.ts

```diff
@@ -360,4 +360,51 @@ export const traceRouter = createTRPCRouter({
         throw e;
       }
     }),
+  deleteMany: protectedProjectProcedure
+    .input(
+      z.object({
+        traceIds: z.array(z.string()),
+        projectId: z.string(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      throwIfNoAccess({
+        session: ctx.session,
+        projectId: input.projectId,
+        scope: "traces:delete",
+      });
+
+      if (input.traceIds.length < 1) {
+        throw new Error("Minimum 1 trace_Id is required.");
+      }
+
+      const traces = await ctx.prisma.trace.findMany({
+        where: {
+          id: {
+            in: input.traceIds,
+          },
+        },
+      });
+
+      if (!traces) {
+        throw new Error("Traces not found in project");
+      }
+
+      return ctx.prisma.$transaction([
+        ctx.prisma.trace.deleteMany({
+          where: {
+            id: {
+              in: input.traceIds,
```

Comment:
Add where condition: projectId -> otherwise user could delete trace of other projects